File: SerenePSM/SerenePSM.Web/Modules/Default/Pfk7121/Pfk7121Columns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace SerenePSM.Default.Columns;

[ColumnsScript("Default.Pfk7121")]
[BasedOnRow(typeof(Pfk7121Row), CheckNames = true)]
public class Pfk7121Columns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public string ColorCode { get; set; }
    public string ColorName { get; set; }
    public string ColorNameS1 { get; set; }
    public string CdColorCategory { get; set; }
    public string ColorPosition { get; set; }
    public string CustomerCode { get; set; }
    public string CheckUse { get; set; }
    public string Remark { get; set; }
    //public string ColorNameS2 { get; set; }
    //public string ColorNameSimple { get; set; }
    
    //public string SeColorBase { get; set; }
    //public string CdColorBase { get; set; }
    //public string SeColorCategory { get; set; }
    
    
    //public string CheckBase { get; set; }
    
    //public string DateInsert { get; set; }
    //public string DateUpdate { get; set; }
    //public string InchargeInsert { get; set; }
    //public string InchargeUpdate { get; set; }
    
    //public string SeSite { get; set; }
    //public string CdSite { get; set; }
}

File: SerenePSM/SerenePSM.Web/Modules/ServerTypes/Default/Pfk7121Columns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Pfk7121Row } from "./Pfk7121Row";

export interface Pfk7121Columns {
    ColorCode: Column<Pfk7121Row>;
    ColorName: Column<Pfk7121Row>;
    ColorNameS1: Column<Pfk7121Row>;
    CdColorCategory: Column<Pfk7121Row>;
    ColorPosition: Column<Pfk7121Row>;
    CustomerCode: Column<Pfk7121Row>;
    CheckUse: Column<Pfk7121Row>;
    Remark: Column<Pfk7121Row>;
}

export class Pfk7121Columns extends ColumnsBase<Pfk7121Row> {
    static readonly columnsKey = 'Default.Pfk7121';
    static readonly Fields = fieldsProxy<Pfk7121Columns>();
}